homeassistant/helpers/trigger.py

```diff
@@ -371,18 +373,99 @@ def entity_filter(entities: set[str]) -> set[str]:
         )
 
 
+class EntityStateTriggerBase(EntityTriggerBase):
+    """Trigger for entity state changes."""
+
+    def is_state_same(self, from_state: State, to_state: State) -> bool:
+        """Check if the old and new states are considered the same."""
+        return from_state.state == to_state.state
+
+    def is_state_to_state(self, state: State) -> bool:
+        """Check if the state matches the target state."""
+        return state.state == self._to_state
+
+    def check_all_match(self, entity_ids: set[str]) -> bool:
+        """Check if all entity states match."""
+        return all(
+            state.state == self._to_state
+            for entity_id in entity_ids
+            if (state := self._hass.states.get(entity_id)) is not None
+        )
+
+    def check_one_match(self, entity_ids: set[str]) -> bool:
+        """Check that only one entity state matches."""
+        return (
+            sum(
+                state.state == self._to_state
+                for entity_id in entity_ids
+                if (state := self._hass.states.get(entity_id)) is not None
+            )
+            == 1
+        )
+
+
+class EntityStateAttributeTriggerBase(EntityTriggerBase):
+    """Trigger for entity state attribute changes."""
+
+    _attribute: str
+
+    def is_state_same(self, from_state: State, to_state: State) -> bool:
+        """Check if the old and new states are considered the same."""
+        return from_state.attributes.get(self._attribute) == to_state.attributes.get(
+            self._attribute
+        )
+
+    def is_state_to_state(self, state: State) -> bool:
+        """Check if the state matches the target state."""
+        return state.attributes.get(self._attribute) == self._to_state
+
+    def check_all_match(self, entity_ids: set[str]) -> bool:
+        """Check if all entity states match."""
+        return all(
+            state.attributes.get(self._attribute) == self._to_state
+            for entity_id in entity_ids
+            if (state := self._hass.states.get(entity_id)) is not None
+        )
+
+    def check_one_match(self, entity_ids: set[str]) -> bool:
+        """Check that only one entity state matches."""
+        return (  # type: ignore[no-any-return]
```

Comment:
mypy complains we might be returning `Any`. I don't understand what the problem is, there's no complaint about the very similar method just above.